lib/segment/tests/integration/payload_index_test.rs

```diff
@@ -611,6 +611,100 @@ fn test_struct_payload_geo_index() {
     }
 }
 
+#[test]
+fn test_struct_payload_geo_polygon_index() {
```

Comment:
This new test is **very** similar to `test_struct_payload_geo_radius_index()` can we _maybe_ extract both into a single function and parametrize it for each test?

Reply:
Didn't find a clean way to parametrize the test. But I split the test to a check fn `test_search_comparison` and make three geo geometry test functions shorter.